Add Type Definitions for react-tag-input

Ogglas opened the pull request:
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

paulvanbrenk commented on types/react-tag-input/index.d.ts:
```diff
@@ -0,0 +1,26 @@
+// Type definitions for React-Tags (react-tag-input) 4.7.2
+// Project: https://github.com/prakhar1989/react-tags
+// NPM: https://www.npmjs.com/package/react-tag-input
```
remove this line, for the Travis CI to pass

paulvanbrenk commented:
Thanks for contributing. 

Please fix the Travis CI build and add a couple of tests to the tests file.

Ogglas commented:
@paulvanbrenk Thanks. I removed the line specified and added a test. However the Travis CI build still fails even though the simple test runs fine on my computer. Can you spot if there is something simple that I have missed?

paulvanbrenk commented on types/react-tag-input/tsconfig.json:
```diff
@@ -0,0 +1,22 @@
+    },
+    "files": [
+        "index.d.ts",
+        "react-tag-input-tests.ts"
```
This should be `react-tag-input-tests.tsx` and add a `"jsx": "preserver"` as a compiler option. The remaining issues should be easy to fix.

Ogglas commented:
@paulvanbrenk Thanks again, should be complete now.
